frontend/src/components/katex.tsx

```
@@ -0,0 +1,29 @@
+  } catch (error) {
+    html = '<span style="color: red;">Invalid formula</span>';
+  }
+
+  return (
+    <div 
+      className="katex-container inline-block align-middle my-0"
+      style={{
+        isolation: 'isolate',
+        contain: 'layout style',
+      }}
+      dangerouslySetInnerHTML={{ __html: html }} 
```

Biome (2.1.2)

[error] Avoid passing content using the dangerouslySetInnerHTML prop.

Setting content using code can expose users to cross-site scripting (XSS) attacks

(lint/security/noDangerouslySetInnerHtml)